### PR TITLE
Fix none feature gates list

### LINKED ISSUE #1456: Init help suggests there are feature gates but lists none
## Is this a BUG REPORT or FEATURE REQUEST?
BUG REPORT

## Versions

**kubeadm version**: `kubeadm version: &version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.4", GitCommit:"c27b913fddd1a6c480c229191a087698aa92f0b1", GitTreeState:"clean", BuildDate:"2019-02-28T13:35:32Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}`

## What happened?
`kubeadm init --help` has this section:
```
[...]
      --feature-gates string                 A set of key=value pairs that describe feature gates for various features. Options are:

  -h, --help                                 help for init
[...]
```
That is, it suggests there are feature gates, but then presents an empty list.

## What you expected to happen?
Looking at [the source](https://github.com/kubernetes/kubernetes/blob/8142ab59046e4535684188968dd135db443ce430/cmd/kubeadm/app/features/features.go#L41), it appears there actually aren't any in the current version. 

The flag should either be omitted completely, or (probably better) explain that there aren't any valid values. I should not have to check the source to understand why there are no options presented.

## How to reproduce it (as minimally and precisely as possible)?
`kubeadm init --help`

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Fix none feature gates list.

**Which issue(s) this PR fixes**:

Fixes kubernetes/kubeadm#1456

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/cmd/config.go [hunk 1]
```diff
@@ -519,7 +519,12 @@ func AddImagesCommonConfigFlags(flagSet *flag.FlagSet, cfg *kubeadmapiv1beta1.In
...
 	)
-	flagSet.StringVar(featureGatesString, "feature-gates", *featureGatesString, "A set of key=value pairs that describe feature gates for various features. "+
-		"Options are:\n"+strings.Join(features.KnownFeatures(&features.InitFeatureGates), "\n"))
+	if knownFeatures := features.KnownFeatures(&features.InitFeatureGates); len(knownFeatures) > 0 {
+		flagSet.StringVar(featureGatesString, "feature-gates", *featureGatesString, "A set of key=value pairs that describe feature gates for various features. "+
+			"Options are:\n"+strings.Join(knownFeatures, "\n"))
+	} else {
+		flagSet.StringVar(featureGatesString, options.FeatureGatesString, *featureGatesString, "A set of key=value pairs that describe feature gates for various features. "+
+			"Options are:\n"+"No gates are available in this release.")
+	}
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/config.go]: i think this `if ...else` block should be made into an utility function here:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/options/generic.go

also for the case where there are no FGs:
it should read:
```
flagSet.StringVar(featureGatesString, options.FeatureGatesString, *featureGatesString, "A set of key=value pairs that describe feature gates for various features. " +
	"No feature gates are available in this release.")
```

**Review by neolit123 — COMMENTED** [review]
thanks for the PR @SataQiu 
added one comment.

/kind cleanup
/priority backlog

File: cmd/kubeadm/app/cmd/config.go (hunk 1 above)
Comment (SataQiu) [reply cmd/kubeadm/app/cmd/config.go]: @neolit123 Thank you very much! I've reorganized the code.

File: cmd/kubeadm/app/cmd/options/generic.go [hunk 2]
```diff
@@ -58,3 +61,14 @@ func AddControlPlanExtraArgsFlags(fs *pflag.FlagSet, apiServerExtraArgs, control
...
 }
+
+// AddFeatureGatesStringFlag adds the --feature-gates flag to the given flagset
+func AddFeatureGatesStringFlag(fs *pflag.FlagSet, featureGatesString *string) {
+	if knownFeatures := features.KnownFeatures(&features.InitFeatureGates); len(knownFeatures) > 0 {
+		fs.StringVar(featureGatesString, FeatureGatesString, *featureGatesString, "A set of key=value pairs that describe feature gates for various features. "+
+			"Options are:\n"+strings.Join(knownFeatures, "\n"))
+	} else {
+		fs.StringVar(featureGatesString, FeatureGatesString, *featureGatesString, "A set of key=value pairs that describe feature gates for various features. "+
+			"No gates are available in this release.")
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/options/generic.go]: one final nit: please change to `No feature gates are available in this release.`

**Review by neolit123 — COMMENTED** [review]
thanks for the update @SataQiu 
added one final comment.

/approve
/hold
@kubernetes/sig-cluster-lifecycle-pr-reviews

**Review by rosti — APPROVED** [review]
Thanks for this @SataQiu !
Please, fix the last nit, pointed by @neolit123 and I'll LGTM it.

**Review by fabriziopandini — APPROVED** [review]
@SataQiu well done!
/approve

**Review by neolit123 — COMMENTED** [review]
thanks
/lgtm
/hold cancel

**Pull request merged**